pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -162,6 +163,15 @@ func (h *NetworkUtilsHandler) ConfigureIpv6Forwarding() error {
 	return err
 }
 
+func (h *NetworkUtilsHandler) IsIpv4Primary() bool {
+	podIP, exist := os.LookupEnv("MY_POD_IP")
+	if !exist {
+		panic(fmt.Errorf("MY_POD_IP doesnt exists"))
```

Comment:
This code is invoked from the virt-handler (on the virt-launcher network namespace). We don't want virt-handler to crash because of an error in a virt-launcher. If you want virt-laucher to be stopped on such an error, just return an error.

Reply:
Petr asked that it would fail, because its a preliminary failure in case someone remove the MY_POD_IP variable and forget its importance.
which in this case, we dont even need to continue the process,
@phoracek 
you suggest to go with the panic or the error ?

Reply:
I missed that this is happening in the handler. As Alona said, this would cause a crash of the whole daemon. Use an error instead.

Reply:
done